src/main/scala/rocks/muki/graphql/codegen/Builder.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 Mediative
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rocks.muki.graphql.codegen
+
+import java.io.File
+import scala.io.Source
+import cats.syntax.either._
+import sangria.parser.QueryParser
+import sangria.validation.QueryValidator
+import sangria.schema._
+import sangria.ast.Document
+
+case class Builder private (
+    schema: Result[Schema[_, _]],
+    document: Result[Document] = Builder.emptyDocumentResult
+) {
+  private def withQuery(query: => Result[Document]): Builder = {
+    val validatedQuery = schema.flatMap { validSchema =>
+      query.flatMap { loadedQuery =>
+        val violations =
+          QueryValidator.default.validateQuery(validSchema, loadedQuery)
+        if (violations.isEmpty)
+          query
+        else
+          Left(Failure(
+            s"Invalid query: ${violations.map(_.errorMessage).mkString(", ")}"))
+      }
+    }
+
+    if (document == Builder.emptyDocumentResult)
+      copy(document = validatedQuery)
+    else
+      copy(document = document.flatMap(doc => validatedQuery.map(doc.merge)))
```

Comment:
That puzzled me in the beginning a lot 😄 
With this we get the "single module" structure in the output, right?

If we would like to build one module per query, we wouldn't need this?